### PR TITLE
Clean up interfaces

### PR DESCRIPTION
**Pull request by OR13**

### CODE REVIEW

**Comment by kimdhamilton** [comment]
Yes, this is what I suspected; just want to make sure we have it documented. Thanks!

**Review by kimdhamilton — APPROVED** [review]

File: docs/interfaces.html [hunk 1]
```diff
@@ -176,6 +136,13 @@ <h3>Issue</h3>
             <li><a href="https://w3c-ccg.github.io/ld-proofs/#dfn-challenge">challenge </a> </li>
             <li><a href="https://w3c-ccg.github.io/ld-proofs/#dfn-proof-type">proofType </a> </li>
         </ul>
+
+        Beware that while the challenge comes from the verifier and must be signed over.
```
Comment (TallTed) [inline docs/interfaces.html:140]:
```suggestion
        Beware that while the challenge comes from the verifier and must be signed over.
```
Should this be an introductory phrase in the sentence that follows below?  If so, some capitalization and punctuation changes are necessary.

Else, is it meant to be a standalone?  In this case, it needs some rewording to actually be a sentence.

**Comment by peacekeeper** [comment]
> focus on refining the higher order APIs

This feels limiting and difficult to maintain for me.. Okay now there is an interface called "Issue" which I supposed is one of those higher order APIs. But how can we anticipate all potential future things we may want to sign? We would always have to change the wallet API just because there's something new we want to sign, no?

But yes I understand there's a tradeoff between flexibility and security/phishing concerns.

WebKMS also seems to have a generic signing function: https://w3c-ccg.github.io/webkms/#sign-options-code-lt-string-gt-code-

**Comment by OR13** [comment]
@peacekeeper yes, I suspect that raw signing will be supported by many lower level interfaces including webkms or other enterprise kms.

I would be happy to add some text that pointed folks who were interested in "raw signatures" to the appropriate specs.

File: docs/interfaces.html (hunk 1 above)
Comment (OR13) [reply docs/interfaces.html:140]: hmm.... for some reason I cannot accept this suggestion...

**Review by mprorock — APPROVED** [review]

**Pull request merged**